domain/src/main/java/camp/nextstep/edu/domain/calculator/ExpressionValidator.kt

```diff
@@ -0,0 +1,33 @@
+package camp.nextstep.edu.domain.calculator
+
+
+class ExpressionValidator {
+
+
+    fun validOrError(expression: List<ExpressionItem>): Boolean {
+        require(isValidHelper(expression)) { "유효하지 않은 표현식입니다." }
+        return true
+    }
+
+    private fun isValidHelper(
+        expression: List<ExpressionItem>
+    ): Boolean {
+        require(expression.isNotEmpty()) { "공백은 입력할 수 없습니다." }
```

Comment:
```suggestion
    fun validOrError(expression: List<ExpressionItem>): Boolean {
        require(expression.isNotBlank()) { "공백은 입력할 수 없습니다." }
        require(isValidHelper(expression)) { "유효하지 않은 표현식입니다." }
        return true
    }
```

validHelper에서 공백체크와 유효하지 않은 표현식 체크가 동시에 이루어질 수 없음에도 불구하고 같은 함수로 묶여져있네요..!
따로 분리해주는 것이 더욱 좋아보입니다.
또한 Empty를 이용하면 " "의 케이스를 체크할 수 없습니다.

Reply:
또한 null 체크 하는 부분도 빠졌네요 😢

Reply:
파라미터의 expression 은 List 타입이라 isNotEmpty() 만으로도 빈 값의 여부를 알 수 있을거 같고, Num.value 가 nonNull int 타입이고 expression 도 nonNull List 라 입력이 들어오는 쪽에서 " " 과 null 을 사용해 입력을 할 수 없을것 같아 밸리데이션에서 제외했습니다!

Reply:
그렇네요!!
그렇다면 함수에서 isNotEmpty만 따로 분리해주면 더욱 좋을 것 같습니다~!